src/ErrorHandling/ErrorState.tsx

```diff
@@ -0,0 +1,35 @@
+import { EventEmitter } from "../utils/EventEmitter";
+
+export type ErrorRecord = {
+  id: number;
+  server: string;
+  errorType: string;
+  message: string;
+  scriptName: string;
+  pid: number;
+  occurrences: number;
+  time: Date;
+  unread: boolean;
+  force?: boolean;
+};
+
+export const ErrorState = {
+  ErrorUpdate: new EventEmitter<[ErrorRecord]>(),
+  ActiveError: null as ErrorRecord | null,
+  Errors: [] as ErrorRecord[],
+  UnreadErrors: 0,
+  PreventModalsUntil: new Date(),
+};
+
+export function errorModalsAreSuppressed(): boolean {
+  return ErrorState.PreventModalsUntil > new Date();
```

Comment:
```suggestion
  return ErrorState.PreventModalsUntil.getTime() > Date.now();
```
Comparing the date objects like you do is still correct, but I think it's better to compare the underlying timestamp values. Comparing the date objects works as long as it does not use equality operators (==, !=, ===, or !==). This means if somebody else changes this line, they may make a mistake (e.g., use `===` to compare the objects).